### PR TITLE
Move Esent support down to Workspaces.Desktop.  This way we use it from RemoteWorkspace

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Note: there is still one VS dependency necessary for our persistence service.  Specifically, we need to know where to actually place the persistence file.  This directory is provided by a VS service.  However, for the out of proc case, this can be something the host passes along so the OOP service has no need to know about VS.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide @heejaechang 

This change moves our Esent code down from ServicesVisualStudio and into Workspaces.Desktop.  The latter project is referenced by RemoteWorkspace and this will help allow us to persist data from the RemoteWorkspace process.

**Comment by heejaechang** [comment]
nice! I guess you will create location service for service hub? not sure where it needs to be saved. probably remotehost need to send location of working folder to remote host so that esent is created in .vs folder for service hub as well. but need to deal with rename of working folder as well.

**Comment by CyrusNajmabadi** [comment]
> probably remotehost need to send location of working folder to remote host so that esent is created in .vs folder for service hub as well. but need to deal with rename of working folder as well.

Right.  My expectation would be that the VS side would remote this information over to the OOP side.

**Comment by CyrusNajmabadi** [comment]
Closed side is tracked here: https://github.com/dotnet/roslyn-internal/pull/1274

Can you take a look @heejaechang

File: src/Workspaces/Core/Desktop/Workspace/SolutionSize/SolutionSizeTracker.cs [hunk 1]
```diff
@@ -32,12 +30,16 @@ internal class SolutionSizeTracker : IIncrementalAnalyzerProvider
         /// </summary>
         public long GetSolutionSize(Workspace workspace, SolutionId solutionId)
         {
-            return workspace is VisualStudioWorkspaceImpl ? _tracker.GetSolutionSize(solutionId) : -1;
+            var service = workspace.Services.GetService<IPersistentStorageLocationService>();
```
Comment (heejaechang) [inline src/Workspaces/Core/Desktop/Workspace/SolutionSize/SolutionSizeTracker.cs]: do we want this in workspace layer or leave it in VS layer? with this in workspace layer, now roslyn used in other place such as ETA will start to use esent.

**Comment by heejaechang** [comment]
:+1:

**Pull request merged**